src/app/mapView/MapView.jsx

```diff
@@ -32,41 +25,109 @@ export default class MapView extends Component {
     navigator.geolocation.getCurrentPosition(
       userPosition => {
         const { coords } = userPosition;
-        this.onCenterChanged({
-          lat: coords.latitude,
-          lng: coords.longitude,
+        this.setState({
+          center : {
+            lat: coords.latitude,
+            lng: coords.longitude,
+          }
         });
       },
       e => console.error('Failed to get current position', e),
       { timeout: geolocationTimeout },
     );
   }
+
   onSearchChanged = event => {
```

Comment:
This should be debounced too.
(Generally any input that triggers as-you-type refreshing of data, without an explicit submit button, should have its API calls debounced.)

Reply:
OK